.github/workflows/release.yml

```diff
@@ -3,14 +3,82 @@ name: Release
 on:
   workflow_dispatch:
 
+
 jobs:
-  gradle:
+  # TOODO how do I ensure that nothing is running on a different commit
+  # 1. We look at the target branch and figure out whether this should be a BUILD
+  # release or a patch release
+  get-update-type:
+    # If the context.ref is refs/heads/main we want to do a standard release
+    # If the context.ref is refs/heads/* we want to do a patch release
+    # If the context.ref is some other ref, we want to fail
+    # Selecting a tag github.ref becomes e.g.: refs/tags/4.1.9.0
+    runs-on: ubuntu-latest
+    permissions:
+      contents: read
+    outputs: # This means this can later be referenced with needs.get-update-type.outputs.update-type
+      update-type: ${{ steps.update-type.outputs.result }}
+    steps:
+      - name: Calculate update type
+        uses: actions/github-script@v7
+        id: update-type
+        with:
+          script: |
+            if (context.ref == "refs/heads/main") {
+                return "BUILD";
+            } else if (context.ref.startsWith("refs/heads/")) {
+                return "PATCH";
```

Comment:
One consequence of this approach is that if someone runs a patch build, they use the patch process that is baked into their branch, rather than the one on main. In some ways, that's good, because it means that if there's a relatively stable release process in a past branch, then past builds will continue to use that. But it also means that we need to be a little mindful of keeping past patch branches up to date if there are any changes that need to be reflected everywhere. Overall, this is probably fine, and there was actually already some dependency between the different branches (as any of the composite actions would borrow the version from the branch), but I did want to raise this nonetheless

Reply:
It also kind of looks like my patch release may be messing with the workflows in general, which seems super weird. 
It, all of a sudden, has 5 runs, and all of them say it's invalid: https://github.com/FoundationDB/fdb-record-layer/actions/workflows/patch_release.yml
I fixed the issue (I think), but nothing happened after that.